quinn-proto/src/connection/assembler.rs

```diff
@@ -132,10 +132,20 @@ impl Assembler {
                 bytes.advance((duplicate.end - offset) as usize);
                 offset = duplicate.end;
             }
+        } else if offset < self.bytes_read {
+            if (offset + bytes.len() as u64) < self.bytes_read {
+                return;
+            } else {
+                let diff = self.bytes_read - offset;
+                offset += diff;
+                bytes.advance(diff as usize);
+            }
```

Comment:
We should get rid of the corresponding read-time code, since this makes it redundant.

Reply:
What code do you think that is, exactly? In my mind the read-time code is more general, since it covers things that are duplicated but still after the current `bytes_read`, so I don't think we can just get rid of it?

Reply:
I was thinking of the `if (chunk.offset + chunk.bytes.len() as u64) <= self.bytes_read` clause, within the ordered case. On review, though, you're correct: if `bytes_read` is 0 and we receive two frames starting at 0, they aren't guaranteed to be deduplicated until read, so that code should be retained.